docs/coding_conventions.md

```diff
@@ -0,0 +1,101 @@
+Coding conventions
+=================
+
+Formatting
+----------
+
+- **Length** of a line should not be longer than 80 characters. In the rare
+  cases when we need a longer line (e.g. in a preprocessor macro) use backslash
+  character to mark a continuation.
+- **Parameter list** is a frequent place, when a line tends to be longer than
+  the limit. Break the line and align parameters, as seen in the example below.
+- **Tab characters** are not welcome in the source. Replace them with spaces.
+- **Indentation** is 2 characters. Two characters are enough to visually
+  emphasize code structure, but do not cause too long lines.
+- **Namespaces** are not indented.
+- **Blocks** Related opening and closing brackets should be placed to the same
+  column (do not follow Java style). This rule holds for namespaces, classes,
+  function blocks and compound statements.
+- **Class declarations** should use only one pulic, protected and private part
+  (in this order). The keywords public, protected, private are not indented,
+  the member declarations are indented as usual (with 2 spaces). Inside a
+  visibility class declare types first, then follow this order: static methods,
+  non-static methods, static attributes, non-static attributes.
+- **Friend** declarations, if any, should be placed before the public
+  visibility items, before the public keyword.
+
+```cpp
+namespace cc
+{
+namespace internal
+{
+
+class Workspace
+{
+  friend void ifAnyFriendExists();
+
+public:
+  Workspace(const std::string& connStr_);
+  ~Workspace();
+
+  void aMethodWithManyParameters(
+    VeryLongParameterItem1 item1_,
+    VeryLongParameterItem2 item2_,
+    Parameter3 par3_);
+
+private:
+  Workspace(const Workspace&); // = delete;
+  Workspace& operator=(const Workspace&); // = delete;
+};
+
+} // internal
+} // cc
+```
```

Comment:
I'd like to see an example source code too, not just header. Especially about things like where `*` and `&` go when parameter has a name (though I personally pretty much **love** the fact that it's tied to the type and not the variable name :purple_heart: ), and how sub-blocks, such as an `if`, a `for` (maybe range-based for too?), an inline lambda, etc. should be used.